src/Bicep.Core/Analyzers/Interfaces/IBicepAnalyzerRule.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Semantics;
+using System.Collections.Generic;
+
+namespace Bicep.Core.Analyzers.Interfaces
+{
+/// <summary>
+/// Implementing IBicepAnalyzer Rule requires
+/// the implementing class to have a parameterless
+/// constructor which can be discoverd through
+/// reflection
+/// </summary>
+    public interface IBicepAnalyzerRule
```

Comment:
Don't like that some of the properties on this interface are user-configurable and some aren't.  Shouldn't there be a separate interface for settings?

Reply:
@StephenWeatherford technically as rules are written we could change everything except "Code" I suppose.  Do you have a proposal? Seems odd to split out an interface to just set DiagnosticLevel.

Reply:
Let's leave for now, we can reevaluate when adding third-party support.